src/config/file/v1.rs

```diff
@@ -64,20 +66,41 @@ impl RawConfigV1 {
     }
 }
 
-fn validate(config: RawConfigV1) -> Result<super::Config, Report<Error>> {
+async fn validate(config: RawConfigV1) -> Result<super::Config, Report<Error>> {
     let endpoint = fossa::Endpoint::try_from(config.endpoint).change_context(Error::Validate)?;
     let key = fossa::Key::try_from(config.integration_key).change_context(Error::Validate)?;
     let api = fossa::Config::new(endpoint, key);
     let debugging = debug::Config::try_from(config.debugging).change_context(Error::Validate)?;
+
+    // Converts and validates the parsed integrations
+    // The first map tries to convert each integration into an integration object after validation
+    // The second map does a best effort approach to inject main/master branch into watched_branches if watched_branches is empty and import_branches is set to true
     let integrations = config
         .integrations
         .into_iter()
         .map(remote::Integration::try_from)
+        .map(|res| async {
+            match res {
+                Ok(integration) => {
+                    Ok(remote::Integration::set_watched_branches(&integration).await)
+                }
+                Err(report) => Err(report),
+            }
+        });
+
+    // Evaluate the reults of the mapping
+    let transformed_integrations = join_all(integrations)
+        .await
+        .into_iter()
+        .map(|res| match res {
+            Ok(integration) => integration,
+            Err(err) => Err(err),
+        })
         .collect::<Result<Vec<_>, Report<remote::ValidationError>>>()
         .change_context(Error::Validate)
         .map(remote::Integrations::new)?;
-
-    super::Config::new(api, debugging, integrations).wrap_ok()
+    println!("the new integrations {transformed_integrations:#?}");
```

Comment:
Should this be debug output?

Reply:
This was supposed to be deleted, good catch